bentoml/yatai/client/interceptor/prom_server_interceptor.py

```diff
@@ -0,0 +1,191 @@
+"""Interceptor a client call with prometheus"""
+
+from timeit import default_timer
+
+import grpc
+from bentoml.yatai.metrics import (
+    GRPC_SERVER_HANDLED_HISTOGRAM,
+    GRPC_SERVER_HANDLED_TOTAL,
+    GRPC_SERVER_STARTED_COUNTER,
+    GRPC_SERVER_STREAM_MSG_RECEIVED,
+    GRPC_SERVER_STREAM_MSG_SENT,
+)
+from bentoml.yatai.utils import (
+    get_method_type,
+    parse_method_name,
+    wrap_interator_inc_counter,
+)
+
+
+def _wrap_rpc_behaviour(handler, fn):
+    """Returns a new rpc handler that wraps the given function"""
+    if handler is None:
+        return None
+
+    if handler.request_streaming and handler.response_streaming:
+        behavior_fn = handler.stream_stream
+        handler_factory = grpc.stream_stream_rpc_method_handler
+    elif handler.request_streaming and not handler.response_streaming:
+        behavior_fn = handler.stream_unary
+        handler_factory = grpc.stream_unary_rpc_method_handler
+    elif not handler.request_streaming and handler.response_streaming:
+        behavior_fn = handler.unary_stream
+        handler_factory = grpc.unary_stream_rpc_method_handler
+    else:
+        behavior_fn = handler.unary_unary
+        handler_factory = grpc.unary_unary_rpc_method_handler
+
+    return handler_factory(
+        fn(behavior_fn, handler.request_streaming, handler.response_streaming),
+        request_deserializer=handler.request_deserializer,
+        response_serializer=handler.response_serializer,
+    )
+
+
+# request are either RPC request as protobuf or iterator of RPC request
+class PromServerInterceptor(grpc.ServerInterceptor):  # pylint: disable=W0232
+    """Interceptor for handling client call with metrics to prometheus"""
```

Comment:
add some related gRPC documentation or example code here for future reference?

e.g.  https://grpc.github.io/grpc/python/grpc.html#service-side-interceptor

Reply:
Should I add examples to documentation under yatai?

Reply:
oh sorry I mean just add it here in the docstring for other BentoML developers